engine/classes/Elgg/Application.php

```diff
@@ -53,6 +56,7 @@ class Application {
 	 */
 	private static $public_services = [
 		//'config' => true,
+		'dic' => true,
 		'menus' => true,
 		'table_columns' => true,
```

Comment:
These named services haven't yet been exposed. Would the idea be to replace the `$public_services` property completely with `elgg-services.php` found from Elgg root?

Reply:
Exposed to the PHP-DI, I mean. So that I could do something like this in my plugin's `elgg-services.php`:
```php
return [
	'test' => DI\object(Elgg\MyPlugin\Test::class)->constructor(Di\get('menus'))),
];
```

Reply:
Well, we could pre-populate with definitions for `\Elgg\Menu\Service`, then you'd put that in your constructor and PHP-DI would auto-wire it.